common/src/main/java/org/conscrypt/ActiveSession.java

```diff
@@ -316,9 +321,13 @@ void onPeerCertificatesReceived(
      */
     void onHandshakeCompleted(String peerHost, int peerPort) {
         id = null;
-        this.peerHost = peerHost;
-        this.peerPort = peerPort;
         this.localCertificates = ssl.getLocalCertificates();
+        if (this.peerCertificates == null) {
+            // When resuming a session, the cert_verify_callback (which calls
+            // onPeerCertificatesReceived) isn't called by BoringSSL during the handshake, leaving
+            // us without the peer certificates.  If that happens, fetch them explicitly.
+            configurePeer(peerHost, peerPort, ssl.getPeerCertificates());
```

Comment:
Consider updating the comment to identify session resumption as the case where this can occur.